system-test/test.sh

```diff
@@ -6,6 +6,18 @@ retry() {
   "${@}" || "${@}" || "${@}" || return 1
 }
 
+function wait_only() {
```

Comment:
Subjective: I'd call it `timeout_after`.

Reply:
That's definitely better!
Done.